java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -2363,13 +2385,26 @@ public CompletableFuture<Map<String, Map<String, String[][]>>> xread(
         return xread(keysAndIds, StreamReadOptions.builder().build());
     }
 
+    @Override
+    public CompletableFuture<Map<GlideString, Map<GlideString, GlideString[][]>>> xreadBinary(
+            @NonNull Map<GlideString, GlideString> keysAndIds) {
+        return xreadBinary(keysAndIds, StreamReadOptions.builder().build());
+    }
+
     @Override
     public CompletableFuture<Map<String, Map<String, String[][]>>> xread(
             @NonNull Map<String, String> keysAndIds, @NonNull StreamReadOptions options) {
         String[] arguments = options.toArgs(keysAndIds);
         return commandManager.submitNewCommand(XRead, arguments, this::handleXReadResponse);
     }
 
+    @Override
+    public CompletableFuture<Map<GlideString, Map<GlideString, GlideString[][]>>> xreadBinary(
+            @NonNull Map<GlideString, GlideString> keysAndIds, @NonNull StreamReadOptions options) {
+        GlideString[] arguments = options.toArgsBinary(keysAndIds);
```

Comment:
Cannot use the `ArgsBuilder `because the `toArgs()` function receives a map argument and parses it in a not trivial way (first adds all the keys and than adds all the values).